manifests/init.pp

```diff
@@ -288,9 +292,24 @@
     require => File[$runtime_dir],
   }
 
-  file { $hints_file:
-    ensure => file,
-    mode   => '0444',
+  # If hints file is not used, Unbound will instead use built-in hints.
+  if $manage_roothints_file {
+    if $skip_roothints_download {
+      File[$hints_file] -> Exec['download-roothints']
+    } else {
+      Exec['download-roothints'] -> File[$hints_file]
+    }
+    if $hints_file_content != undef and $skip_roothints_download {
+      $_hints_file_content = { 'content' => $hints_file_content }
+    } else {
+      $_hints_file_content = {}
+    }
+
+    file { $hints_file:
+      ensure => file,
+      mode   => '0444',
+      *      => $_hints_file_content,
+    }
   }
```

Comment:
Its fine to pass `content => undef` as such i think we can simplify this to: 
```suggestion
    $_hints_content = $skip_roothints_download ? {
      true      => $hints_file_content,
      default => undef,
    }
    file { $hints_file:
      ensure => file,
      mode   => '0444',
      content => $_hints_file_content,
    }
```

or 
```suggestion
if $skip_roothints_download {
  File[$hints_file] {
    content => $hints_file_content
  }
}
```

Reply:
I like that approach and have adjusted it per your recommendation.